akka-contrib/src/test/scala/akka/contrib/circuitbreaker/sample/CircuitBreakerAskWithFailure.scala

```diff
@@ -0,0 +1,38 @@
+package akka.contrib.circuitbreaker.sample
+
+import akka.actor.{ Actor, ActorLogging, ActorRef }
+import akka.contrib.circuitbreaker.CircuitBreakerActor._
+import akka.util.Timeout
+
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+//#ask-with-failure-sample
+class CircuitBreakerAskWithFailure(potentiallyFailingService: ActorRef) extends Actor with ActorLogging {
+  import SimpleService._
+  import akka.pattern._
+
+  implicit val askTimeout: Timeout = 2.seconds
+
+  val serviceCircuitBreaker =
+    context.actorOf(
+      CircuitBreakerActorBuilder(maxFailures = 3, callTimeout = askTimeout, resetTimeout = 30.seconds).propsForTarget(potentiallyFailingService),
+      "serviceCircuitBreaker")
+
+  import context.dispatcher
+
+  override def receive: Receive = {
+    case AskFor(requestToForward) ⇒
+      (serviceCircuitBreaker ? Request(requestToForward)).failForOpenCircuit.mapTo[String].onComplete {
```

Comment:
I'm going to try a couple of things. One is to extract the `failForOpenCircuit` enabling implicit in a different object since I agree with you that current way is a bit obscure.
The second thing I was thinking about was to have a method `withCircuitBreaker` (or `failForOpenCircuit` for consistency) to use around the future with this syntax:

```(scala)
withCircuitBreaker {  serviceCircuitBreaker ? Request(requestToForward) }.mapTo[String].onComplete {
```